tests/test_dependency_rendering.py

```diff
@@ -0,0 +1,218 @@
+from .testutils import create_and_process_template_response, Django30CompatibleSimpleTestCase as SimpleTestCase
+
+
+class SimpleComponentAlternate(component.Component):
+    def context(self, variable):
+        return {}
+
+    def template(self, context):
+        return "simple_template.html"
+
+    class Media:
+        css = {"all": ["style2.css"]}
+        js = ["script2.js"]
+
+
+class SimpleComponentWithSharedDependency(component.Component):
+    def context(self, variable, variable2="default"):
+        return {
+            "variable": variable,
+            "variable2": variable2,
+        }
+
+    def template(self, context):
+        return "simple_template.html"
+
+    class Media:
+        css = {"all": ["style.css", "style2.css"]}
+        js = ["script.js", "script2.js"]
+
+
+class MultistyleComponent(component.Component):
+    def template(self, context):
+        return "simple_template.html"
+
+    class Media:
+        css = {"all": ["style.css", "style2.css"]}
+        js = ["script.js", "script2.js"]
+
+
+class ComponentMediaRenderingTests(SimpleTestCase):
+    def setUp(self):
+        # NOTE: component.registry is global, so need to clear before each test
+        component.registry.clear()
+
+    def assert_script_count(self, rendered, script_name, count):
+        """Assert that named script is included in rendered HTML in either old or new Django format.
+
+            New format: '<script src="{}"></script>'
+            Old format: '<script type="text/javascript" src="{}"></script>'"""
+        try:
+            self.assertInHTML('<script src="{}"></script>'.format(script_name), rendered, count=count)
+        except AssertionError:
+            self.assertInHTML('<script type="text/javascript" src="{}"></script>'.format(script_name),
+                              rendered, count=count)
+
+    def assert_stylesheet_count(self, rendered, stylesheet_name, count):
+        """Assert that named stylesheet is included in rendered HTML in either old or new Django format.
+
+            New format: '<link href="{}" type="text/css" media="{}" rel="stylesheet">'
+            Old format: '<link href="{}" type="text/css" media="{}" rel="stylesheet">'"""
+        try:
+            self.assertInHTML('<link href="{}" type="text/css" media="all" rel="stylesheet">'.format(stylesheet_name),
+                              rendered, count=count)
+        except AssertionError:
+            self.assertInHTML(
+                '<link href="{}" type="text/css" media="all" rel="stylesheet" />'.format(stylesheet_name),
+                rendered, count=count)
+
+    def test_no_dependencies_when_no_components_used(self):
+        component.registry.register(name="test", component=SimpleComponent)
+
+        template = Template("{% load component_tags %}{% component_dependencies %}")
+        response = create_and_process_template_response(template)
+        rendered = response.content.decode('utf-8')
```

Comment:
All tests currently decode the content which feels somewhat repetitive. Does it make sense to move it inside create_and_process_template_response?

Reply:
Done.